clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -4111,7 +4111,10 @@ static ActionResult<CXXRecordDecl *> getPatternForClassTemplateSpecialization(
         if (Ambiguous) {
           // Partial ordering did not produce a clear winner. Complain.
           Inst.Clear();
-          ClassTemplateSpec->setInvalidDecl();
+
+          if (!S.isSFINAEContext())
+            ClassTemplateSpec->setInvalidDecl();
```

Comment:
Are we still getting any benefit from marking this declaration as invalid in the remaining case?

Reply:
> The question is, Is this declaration specially malformed so that we need to mark it as invalid, in order to have an easier time dealing with it downstream?

I was concerned about uses like `CodeGeneratorImpl::HandleTopLevelDecl()` but that actually looks at whether an unrecoverable error occurred, not whether the top-level decl was marked invalid. So it's possible I was guarding against something that's not an issue.

Reply:
Can we make a decision here?

Reply:
In the absence of a concrete reason to do this, I'd say it's fine to remove the `setInvalidDecl()` entirely as you did originally. Sorry for the noise!